packages/core/src/scope.ts

```diff
@@ -0,0 +1,126 @@
+import { Scope as BaseScope } from '@sentry/shim';
+import { Breadcrumb, SentryEvent, User } from '@sentry/types';
+
+/** An object to call setter functions on to enhance the event */
+export class Scope implements BaseScope {
+  /**
+   * Create a new empty internal scope. This will not be exposed to the user.
+   *
+   * @param breadcrumbs
+   * @param user
+   * @param tags
+   * @param extra
+   * @param fingerprint
+   * @param scopeChanged
+   */
+  public constructor(
+    public breadcrumbs: Breadcrumb[] = [],
```

Comment:
A couple of nit picks related to this constructor:
 - Can we please make these fields and assign them explicitly? I know this is more code, but we use that consistently throughout the code base.
 - Function arguments are position dependent and do not scale as nicely. If you really need to pass scope data here, I'd propose to put a typed `data` object as optional function argument
 - The `scopeChanged` parameter seems strange in this position. It might be nice(r) to just leave the funciton.